### PR TITLE
BEP Describe personas and separate portal and framework docs

### PR DESCRIPTION
**Pull request by waldirmontoya25**
## Hey, I just made a Pull Request!

BEP Describe Personas associated with a Backstage adoption journey and propose the clear separation of the Backstage Developer Portal and Backstage framework docs.

### CODE REVIEW

File: beps/0006-docs-personas-framework-portal/README.md [hunk 1]
```diff
@@ -0,0 +1,240 @@
...
+   - **Administrator/Operator**: A person who configures, secures, and deploys the developer portal, manages plugins, and oversees the general administration of the developer portal.
+   - **Integrator/Builder**: A person who builds plugins, customizes the code and design, and creates custom-built developer portals based on the Backstage framework. This includes developers and designers and anyone adding new functionality to their own Backstage instance.
+   - **Product Manager/Business stakeholders**: A person who defines the strategy for adopting Backstage, identifies use cases, communicates the value proposition for adopting Backstage and connects the developer portal to the business strategy.
+   - **Contributor**: A person who contributes to the Backstage upstream ecosystem.
+
+The adoption strategy would be as follows:
+
+- Create a dedicated page describing the personas Backstage is targeting and the documentation sections that cater to each persona.
+- Restructure the documentation into the two new sections (Framework and Developer Portal) and redistribute the existing content accordingly.
```
Comment (aramissennyeydd) [inline beps/0006-docs-personas-framework-portal/README.md:67]: What are we scoping as "restructuring" or "redistributing"? Is this primarily just moving the existing docs into the separate ToCs? While working on #22541, I noticed a lot of documentation that is either out of date, doesn't have a great narrative within or doesn't fit into the overall narrative for the section.

**Review by aramissennyeydd — COMMENTED** [review]
🙌 thanks for getting this moving!!

File: beps/0006-docs-personas-framework-portal/README.md (hunk 1 above)
Comment (waldirmontoya25) [reply beps/0006-docs-personas-framework-portal/README.md:67]: I think it is a mix of things, but I found it too hard to describe it in detail in a single BEP. But more than open to any suggestions.

For this one specifically, I thought about the following:

1. Aligning the concepts related to Personas, Backstage Developer Portal, and Framework to Build Developer Portals.
2. Dividing the Docs in two sections: Developer Portal and Framework
3. Updating overviews and descriptions according to the proposed changes.
4. Updating the table of contents, taking into considerations the proposed Personas and the CNCF [recommendation](https://github.com/backstage/backstage/issues/21946) 
5. Redistributing content according to the new structure

Next steps:

6. Completing the updates based on the [CNCF ](https://github.com/backstage/backstage/issues/21893)recommendations
7. Renaming sections to standardizing the content
8. Review each individual page to update and correct content

Comment (aramissennyeydd) [reply beps/0006-docs-personas-framework-portal/README.md:67]: I'm a bit torn here, I like the idea of separating the documentation into the two separate sections you describe, but I don't think changing the ToC is where we should start, especially if outputs of this update are going to be an announcement of the changes. IMO, we should align on narrative throughout the documentation and general parameters for how to write documentation for specific user profiles first before diving in to updating the organization. That would give us a shared understanding of what each role means concretely and lay the foundation for organization or content updates.

Priorities in my mind:
1. Create a documentation narrative overview
2. Create guidelines for writing docs for Admins/Devs/Integrators
3. Start condensing/adjusting existing documentation to the guidelines above and relocate in the existing ToC.
4. Decide where to split the documentation and do it in a single PR or small set of PRs.
5. Announce.

I'm definitely out of touch with the current adopter experience, but I don't quite understand the focus on differentiating framework and developer portal rn, that feels like something nuanced enough that we should tackle it specifically in the relevant sections -- most likely overview.

tldr; I think we should start from the bottom-up instead of top-down, starting with a set of guidelines for narrative and voices for each of the roles, then working to update the current table of contents organically.

Comment (waldirmontoya25) [reply beps/0006-docs-personas-framework-portal/README.md:67]: I follow you @aramissennyeydd, and I think I understand your perspective. 

Defining a clear top-down structure is, in my opinion, crucial before making any significant changes to the documentation. Without a well-defined structure, it can be confusing and challenging to determine what changes to make and how to comfortably relocate content. 

Although I initially proposed separating the docs into two products – a Developer Portal and a Framework to Build Developer Portals – a similar outcome could be accomplished by reorganizing the Table of Contents (ToC) as described by the CNCF [**recommendation**](https://github.com/backstage/backstage/issues/21946). Following a similar approach, WordPress clearly separates its [**general documentation**](https://wordpress.org/documentation/) from its [**framework documentation**](https://developer.wordpress.org/).

To clarify, the proposal's intention is to define the personas involved in the Backstage adoption journey and to remove adoption barriers by differentiating the content targeting developers, so that new adopters, assessing, trying, deploying, and using Backstage can follow content that doesn't require deep knowledge of the Backstage codebase to understand what it is, what problem it solves, and can start realizing value from it. This could be accomplished by separating the docs into "two products" or by separating the Developer/Integrator content into its own section, as [**proposed**](https://github.com/backstage/backstage/issues/21946) by the CNCF.

I believe clearly defining the user personas, outlining the structure of the ToC, and relocating content accordingly is a critical step to identify the gaps, given that the current docs have mixed content for admins, developers/integrators, users, and business guides while heavily focused on the Developer persona. This is, IMO why most of the [**general recommendations**](https://github.com/backstage/backstage/issues/21893) are related to creating Admin guides. However, plenty of content exists related to Admin guides in the existing docs but is mixed with Developer-oriented content, which is why restructuring the ToC would guide us on what content to move where. Just my two cents.

Comment (aramissennyeydd) [reply beps/0006-docs-personas-framework-portal/README.md:67]: Alright, I'm following, I think we're talking in pretty similar directions. I'd like to add a more prescriptive description of each of the user roles and each high level ToC item, that's the missing piece for me right now. Relocating and updating the documentation in flight makes sense, but I'd like to know more holistically why we're putting certain docs in certain places. I'm at BackstageCon so it'll be a little bit of turnaround, might not be able to get to this until tonight.

Comment (aramissennyeydd) [reply beps/0006-docs-personas-framework-portal/README.md:67]: Comparing across forks on GH is a little weird, ended up opening https://github.com/waldirmontoya25/backstage/pull/1 with possible updates.

**Comment by benjdlambert** [comment]
@waldirmontoya25 apologies - but we already merged a BEP-0006 😅  We're going to run out of numbers soon. Could you update this to be 0007 instead? :pray:

**Comment by waldirmontoya25** [comment]
> @waldirmontoya25 apologies - but we already merged a BEP-0006 😅 We're going to run out of numbers soon. Could you update this to be 0007 instead? 🙏
> 
> Side note, I wonder if there's a better process for this, maybe the number could be added in a seperate PR which is merged quicker than it being in a pending approval state.

hey @benjdlambert sure, I just removed the numbers 😅

File: beps/0007-docs-personas-framework-portal/README.md [hunk 2]
```diff
@@ -0,0 +1,347 @@
...
+- Adoption use cases
+- Adoption strategies
+- Measuring success
+- Case studies
+
+## Release Plan
+
+- Release the BEP by 03/24/2024.
+- Discuss the changes with the community and gather feedback by 04/24/2024.
+- Implement the changes by 04/30/2024.
```
Comment (aramissennyeydd) [inline beps/0007-docs-personas-framework-portal/README.md:180]: Can we adjust this timeline a bit? 6 days for implementation feels intense even if this is just scoped to ToC adjustments.

File: beps/0007-docs-personas-framework-portal/README.md [hunk 3]
```diff
@@ -0,0 +1,347 @@
...
+- Write additional content beyond the scope of the existing documentation
+
+## Proposal
+
+The proposed restructuring of the Backstage documentation revolves around two core ideas:
+
+1. Positioning Backstage as both a framework to build developer portals and a developer portal itself, and splitting the documentation into two main sections:
+
+   - Developer Portal: Focusing on the features, configuration, and usage of the developer portal that users get out of the box with the core plugins.
+   - Framework: Covering the aspects of Backstage as a framework, including guides for integrators and builders who want to create their own developer portal using Backstage.
```
Comment (aramissennyeydd) [inline beps/0007-docs-personas-framework-portal/README.md:66]: Should we add a similar style as for personas for each of these documentation types? Are we separating these out to just personas, ie Developer Portal is User/Administrator/Product Manager and Framework is Integrator/Contributor? Or is there more nuance? Do we want users to refer to both documentation types depending on their role? ex: Should administrators feel empowered to look through the framework docs even though it's not super relevant to their day to day?

**Review by aramissennyeydd — APPROVED** [review]
I said I had more comments from my initial review, but this is looking good 😁 !

In BEP style, let's keep iterating on this, I like this initial draft and I have a few ideas for follow up PRs.

Off to the maintainers 🚢 .

File: beps/0007-docs-personas-framework-portal/README.md [hunk 4]
```diff
@@ -0,0 +1,347 @@
...
+
+None
+
+## Example Table of Contents
+
+- Overview
+  - "The overview should introduce users to the concept of Backstage, what an IDP is, how to deliver value, why you should care about DevEx, etc."
+  - What is Backstage?
+  - Roadmap
+  - Vision
```
Comment (dwelsch-esi) [inline beps/0007-docs-personas-framework-portal/README.md:192]:
```suggestion
- Overview
  - "The overview should introduce users to the concept of Backstage, what an IDP is, how to deliver value, why you should care about DevEx, etc."
  - What is Backstage?
  - Vision
  - Roadmap
  - User personas and use cases
  - How this documentation is organized
    - Framework
    - Developer portal
```

**Review by dwelsch-esi — COMMENTED** [review]
Awesome plan for reorganizing the documentation! My only overall comment is to keep in mind the goal of making the users successful by writing task-oriented documentation. Good stuff!

**Comment by efenner-cambia** [comment]
bump

File: beps/0007-docs-personas-framework-portal/README.md [hunk 5]
```diff
@@ -0,0 +1,347 @@
...
+- Configuring
+  - Authentication
+  - Plugins
+- Ingesting data (users/groups/components, etc)
+- Installing plugins
+- Implement Git Flows for the Developer portal
+- Creating Pipelines for Docs generation
+- Troubleshooting
+
+#### Integrator
```
Comment (Rugvip) [inline beps/0007-docs-personas-framework-portal/README.md:133]: Tbh I'm not sure the split between admin and integrator is all that useful. My thinking here is that we should merge them and be upfront about the fact that an admin/integrator needs to do some coding to set up Backstage. Over time we strive to remove that, having it become more of "advanced customization" sections where we generally link over to builder docs for the actual instructions instead. What do you think?

File: beps/0007-docs-personas-framework-portal/README.md [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+
+- Release the BEP by 03/24/2024.
+- Discuss the changes with the community and gather feedback by 04/24/2024.
+- Implement the changes by 04/30/2024.
+
+## Dependencies
+
+None
+
+## Example Table of Contents
```
Comment (Rugvip) [inline beps/0007-docs-personas-framework-portal/README.md:186]: 👍, thinking a good process to settle this BEP is to first agree on the pieces above, and once we've done that to dive into this section

**Review by Rugvip — CHANGES_REQUESTED** [review]
Thank you! 🎉 

Sorry it's taken a long time to get to this one 🙈

Good to go and very much something we want to prioritize and work on! 👍 Just needs frontmatter update + 0007 -> 0008 to be able to merge.

I do want to follow up with some different suggestions about the breakdown though, might have a slightly different take on the portal/framework split. Will handle that in followup PRs with separate discussions though

File: beps/0007-docs-personas-framework-portal/README.md (hunk 5 above)
Comment (dwelsch-esi) [reply beps/0007-docs-personas-framework-portal/README.md:133]: @Rugvip The distinction between user roles is to help organize the task documentation. The main question should be: What is the relationship between the set of tasks (procedures, processes) that the admin needs to do vs. what the integrator needs to do? If the two sets are largely disjoint, then they should be separate personas. If there's some overlap, it's a judgement call.

My intuition is that the tasks associated with creating a plugin can be self-contained; however, if they're mostly going to be undertaken by admins, then I agree with your arguments for combining the roles. If that's the case, then maybe consider making the the integrator documentation a chapter/section of the admin documentation? That would maintain the separation of the plugin-creation documentation (if I'm creating a plugin, I want all those resources in one place) while making it clear that this is an undertaking for an admin or admin-adjacent persona.

Another advantage to this approach (integration as a subsection of admin) is that it gives you the flexibility to move things around later if/when (as you suggest above) you relieve the admin of some of the coding requirements.

File: beps/0007-docs-personas-framework-portal/README.md (hunk 2 above)
Comment (dwelsch-esi) [reply beps/0007-docs-personas-framework-portal/README.md:180]: @aramissennyeydd @waldirmontoya25 Agreed. This is going to require more than a week even if you're able to reorganize the documentation without writing or revising a word of it (which is doubtful as well). I'd think of this as two- or three-week sprint for one contributor. Keep in mind that review and discussion will push out calendar time even if someone is able to get the work done quickly.

File: beps/0007-docs-personas-framework-portal/README.md (hunk 5 above)
Comment (Rugvip) [reply beps/0007-docs-personas-framework-portal/README.md:133]: @dwelsch-esi It makes sense to me if we define "integrator" and the strict subset of the "admin" persona that needs to do any form of coding. Essentially a persona that we're looking to remove long-term. It also makes sense from a documentation maintenance point of view, since we can eventually just delete that content. My main worry is that it's not going to be a good structure for the docs in the short to medium term, as it might be that the two personas are currently too intertwined.

Plugin creation would be strictly the "builder" persona and I think we should keep that separate from the other two.

Comment (dwelsch-esi) [reply beps/0007-docs-personas-framework-portal/README.md:133]: @Rugvip So are you proposing separating the Builder and Integrator into two different personas? Looking back through the BEP, I see both names are used together ("Integrator/Builder"). My recommendation: If they're the same, pick one name and stick with it. If they're distinct, separate and define them.

**Review by Rugvip — APPROVED** [review]
Thank you! I moved it over to 0008, merged a few suggestions, and gonna ship it. Wanted to make sure people are able to contribute to this BEP through separate PRs.
